packager/packager.cc

```diff
@@ -471,11 +608,141 @@ unique_ptr<PackageInfo> getPackageInfo(core::MutableContext ctx, ast::ParsedFile
     return move(finder.info);
 }
 
+// For a given package, a tree that is the union of all constants exported by the packages it
+// imports.
+class ImportTree final {
+    struct Source {
+        core::NameRef packageMangledName;
+        core::LocOffsets importLoc;
+        bool exists() {
+            return importLoc.exists();
+        }
+    };
+
+    // To avoid conflicts, a node should either be a leaf (source exists, no children) OR have children
+    // and an non-existent source. This is validated in `makeModule`.
+    UnorderedMap<core::NameRef, std::unique_ptr<ImportTree>> children;
+    Source source;
+
+public:
```

Comment:
Why make this private? It's already in an anonymous namespace, so it won't leak out beyond this file.

Reply:
The location of this comment is a bit misleading, I meant to ask why the map and source were private :)

Reply:
While true, I think this highlights the expected use better.